polkadot/xcm/pallet-xcm/src/lib.rs

```diff
@@ -1495,13 +1500,19 @@ impl<T: Config> Pallet<T> {
 		let outcome =
 			T::XcmExecutor::execute_xcm_in_credit(origin, local_xcm, hash, weight, weight);
 		Self::deposit_event(Event::Attempted { outcome: outcome.clone() });
-		if let Some(remote_xcm) = remote_xcm {
-			outcome.ensure_complete().map_err(|_| Error::<T>::LocalExecutionIncomplete)?;
+		outcome.ensure_complete().map_err(|error| {
+			log::error!(target: "runtime::xcm", "XCM executio failed with error {:?}", error);
+			Error::<T>::LocalExecutionIncomplete
+		})?;
 
+		if let Some(remote_xcm) = remote_xcm {
 			let (ticket, price) = validate_send::<T::XcmRouter>(dest, remote_xcm.clone())
 				.map_err(Error::<T>::from)?;
 			if origin != Here.into_location() {
-				Self::charge_fees(origin, price).map_err(|_| Error::<T>::FeesNotMet)?;
```

Comment:
someone should make a rule to disallow ignoring errors

Reply:
The error is clearly handled... What you want is "better observability".

Reply:
yeah maybe I wasn’t using the right words but the original error details are ignored and that makes debugging hard. I have been bite by this kind of code too many times and I just don’t want to see it anymore